### DISCUSSION TITLE
@rtk-query/codegen-openapi and transformResponse

### DISCUSSION THREAD

**Discussion by nhayfield** [reduxjs/redux-toolkit#3506] [Q&A] [answered]
I am generating from a large openapi.yaml file using codegen-openapi. It works well. 
However some of the list methods can return 1000s of results in an array. I want to use transformResponse and createEntityAdapter for them for faster lookups. 
I can't seem to figure out how to make it work with the generated file. 
I see there is the endpointOverrides?: EndpointOverrides[] option when generating but it doesn't seem to support transformResponse. 
Oh yeah, I am using Typescript too. :)

**Comment by EskiMojo14** [comment, marked as answer]
you would need to use [`enhanceEndpoints`](https://redux-toolkit.js.org/rtk-query/api/created-api/code-splitting#enhanceendpoints).

### 1.9.4+
currently changing the ResultType of an endpoint is fairly complicated:
```ts
import type { DefinitionsFromApi, TagTypesFromApi, OverrideResultType } from "@reduxjs/toolkit/dist/query/endpointDefinitions"
import { api } from "./autogenerated"

type Definitions = DefinitionsFromApi<typeof api>
type TagTypes = TagTypesFromApi<typeof api>

type UpdatedGetPostsDef = OverrideResultType<Definitions["getPosts"], EntityState<Post>>

type UpdatedDefinitions = Omit<Definitions, { getPosts: UpdatedGetPostsDef }>

const enhanced = api.enhanceEndpoints<TagTypes, UpdatedDefinitions>({
  endpoints: {
    getPosts: {
      transformResponse: (posts) => postAdapter.setAll(postAdapter.getInitialState(), posts)
    }
  }
})
```

There's also an open [issue](https://github.com/reduxjs/redux-toolkit/issues/3443) for this inference not working correctly, which there is an open [PR](https://github.com/reduxjs/redux-toolkit/pull/3500) for.

### 2.x
In 2.0, we're investigating replacing enhanceEndpoints completely, with separate `addTagTypes` and `enhanceEndpoint` functions - this will greatly simplify inference for use cases such as overriding ResultType:
```ts
const enhanced = api.enhanceEndpoint('getPost', {
  transformResponse: (posts) => postAdapter.setAll(postAdapter.getInitialState(), posts)
})
```
The PR for this is https://github.com/reduxjs/redux-toolkit/pull/3485.

**Reply by andreyantipov** [reply]
Could you please update example for 1.9.4+?

```ts
import type { DefinitionsFromApi, TagTypesFromApi } from "@reduxjs/toolkit/query"
```

to:
```ts
import type { DefinitionsFromApi, TagTypesFromApi, OverrideResultType } from "@reduxjs/toolkit/dist/query/endpointDefinitions"
```

based on this source files and PR its not available directly from index file:
https://github.dev/reduxjs/redux-toolkit
https://github.dev/reduxjs/redux-toolkit/pull/2953/files#diff-0488a200c74c32442afa991afb0a92f96df8ff2d20ac74ad7bc76d8d701a59db

**Reply by EskiMojo14** [reply]
yep, updated - wasn't aware that they weren't exported from the index.

**Comment by nhayfield** [comment]
thank you for the help. is there an eta on the 2.0 version? in a few months?

**Reply by EskiMojo14** [reply]
No solid timeline as far as I know - though it's recently hit beta so we probably won't want to leave it for too long.

wrt the PR specifically it just depends when my fellow maintainers are available to review it 🙂

**Comment by nhayfield** [comment]
fairr enough.
